scipy/optimize/_numdiff.py

```diff
@@ -370,6 +386,12 @@ def approx_derivative(fun, x0, method='3-point', rel_step=None, abs_step=None,
         a 2-D structure. For ndarrays, if m=1 it is returned
         as a 1-D gradient array with shape (n,).
 
+    info_dict : dict
```

Comment:
`_ScalarFunctionWrapper` can't be relied on to keep track of number of function evaluations in a parallel context. When parallelisation is employed the object is recreated in a new Process, and aggregating `nfev` is pretty difficult. `approx_derivative` can track the number of evaluations (it's the one controlling the parallel evaluation), so report the `nfev` if requested.